posthog/temporal/data_imports/pipelines/meta_ads/source.py

```diff
@@ -0,0 +1,430 @@
+        else:
+            yield from _get_resource_data(config.account_id, schema, config.resource_name, access_token)
+
+    return SourceResponse(
+        name=name,
+        items=get_rows(),
+        primary_keys=schema.primary_key,
+        partition_count=1,
+        partition_size=1,
+        partition_mode="datetime",
+        partition_format="month",  # the api is daily so bundle into months
+        partition_keys=["date_start"] if is_stats else ["created_time"],
+    )
+
+
+def _get_insights_data(
+    account_id: str,
+    schema: MetaAdsSchema,
+    time_range: dict | None,
+    access_token: str,
+) -> collections.abc.Generator[list[dict], None, None]:
+    """Get insights data from Meta Ads API."""
+
+    params = {
+        "fields": ",".join(schema.field_names),
+        "level": schema.name,
+        "time_increment": 1,
+        "limit": 100,
+    }
+
+    try:
+        url = f"https://graph.facebook.com/{API_VERSION}/{account_id}/insights"
+        rows = []
+
+        for page_data in _make_paginated_api_request(url, params, access_token, time_range):
+            for insight in page_data.get("data", []):
+                row_data = {}
+                for k, v in insight.items():
+                    row_data[k] = _serialize_value(v)
+
+                rows.append(row_data)
+
+                if len(rows) >= DEFAULT_CHUNK_SIZE:
+                    yield rows
+                    rows = []
+
+        if rows:
+            yield rows
+
+    except Exception as e:
+        capture_exception(e)
+        raise
+
+
+def _get_resource_data(
+    account_id: str,
+    schema: MetaAdsSchema,
+    resource_name: str,
+    access_token: str,
+) -> collections.abc.Generator[list[dict], None, None]:
+    """Get regular resource data from Meta Ads API."""
+    field_names = [field for field in schema.field_names if field != "account_id"]
+
+    try:
+        endpoint = RESOURCE_ENDPOINTS.get(MetaAdsResource(resource_name))
+        if endpoint is None:
+            raise ValueError(f"Unknown resource: {resource_name}")
+
+        if resource_name == "account":
+            url = f"https://graph.facebook.com/{API_VERSION}/{account_id}"
+        else:
+            url = f"https://graph.facebook.com/{API_VERSION}/{account_id}/{endpoint}"
+
+        params = (
+            {
+                "fields": ",".join(field_names),
+                "limit": 100,
+            }
+            if field_names
+            else {}
+        )
+        rows = []
+
+        if resource_name == "account":
+            data = _make_api_request(url, params, access_token)
+            resources = [data]
+
+            for resource in resources:
+                row_data = {}
+
+                for field in field_names:
+                    value = resource.get(field)
+                    row_data[field] = _serialize_value(value)
+
+                rows.append(row_data)
+
+            if rows:
+                yield rows
```

Comment:
I don't really understand whats going on here? We're making `data` into an array (`[data]`) and then iterating over it?? Why? Does the `account` endpoint only return a single row?

Reply:
We could just yield the whole object instead?